### ISSUE TITLE
[Feature] support synchronize multiple table into multiple paimon table

### ISSUE THREAD

**Issue by zhangjun0x01** [apache/incubator-paimon#1248]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Now, users can synchronize one or multiple  tables from MySQL into one Paimon table. I think it is  necessary to synchronize multiple MySQL tables to the corresponding paimon table

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by tsreaper** [comment]
Hi @zhangjun0x01 , thanks for opening this issue.

However I can't quite understand what you mean. What's the difference between "synchronize one or multiple tables from MySQL into one Paimon table" and "synchronize multiple MySQL tables to the corresponding paimon table"? To me they are the same.

**Comment by zhangjun0x01** [comment]
hi, @tsreaper 

"synchronize one or multiple tables from MySQL into one Paimon table"

  mysql : db1.t1 + db2.t2 --> paimon : db. t3

"synchronize multiple MySQL tables to the corresponding paimon table"

  mysql : db1.t1 + db2.t2 --> paimon : db1.t1 + db2.t2 


the first case : we can synchronize multiple mysql table to one paimon table , build a wide-table on paimon.

the second case : There is no relationship between the MySQL tables, so I cannot merge them into one paimon table. I want to use one Flink job to synchronize all MySQL tables to the corresponding paimon table, instead of synchronizing one table for each Flink job, so that reduce resource consumption.

**Comment by qidian99** [comment]
Yes this feature is applicable in some scenarios where the user physically splits the table into multiple tables (either vertically or horizontally), and tries to merge them into one Paimon table.

Some additionally fields are needed to deal with primary key conflicts and data lineage.

**Comment by gfunc** [comment]
This also applies to CDC Kafka, IMO it is better if we assign partition keys with values extracted from Kafka meta (topic etc.)/canal meta(database etc.).
